main_test.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"os/exec"
+	"testing"
+)
+
+func TestStartServer(t *testing.T) {
```

Comment:
putting it here - add `test` target to Makefile and a github action to test in PRs. Unless configuting a github action to do that is too much work, in which case maybe just add a checkbox to the PR for the reviewer to run `make test` locally before approving the PR.

Reply:
added a `test` target to the make file.  I think I am going to hold off in the github action.  Setting that up would be some scope creep.  But we can add a card to add some of that if we want to.

WRT the checkbox, there was no PR template, so I added one and added a checkbox.